pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -94,29 +93,47 @@ func (c *Reconciler) reconcileDeployment(ctx context.Context, rev *v1.Revision)
 				}
 			}
 
+			// if a Pod is terminating already, we do not care it being not ready because this is expected to be the case
+			if pod.DeletionTimestamp != nil {
+				continue
+			}
+
+			// if a Pod is ready, then do not check it. The fact it is ready may not yet be reflected on the Deployment status.
+			for _, condition := range pod.Status.Conditions {
+				if condition.Type == corev1.PodReady && condition.Status == corev1.ConditionTrue {
+					continue podsLoop
+				}
+			}
```

Comment:
Why do we need this part? If there are no available replicas nothing will be ready, no?